metrics_sdk/lib/opentelemetry/sdk/metrics/export/in_memory_metric_pull_exporter.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module Export
+        # The InMemoryMetricPullExporter behaves as a Metric Reader and Exporter.
+        # To be used for testing purposes, not production.
+        class InMemoryMetricPullExporter
+          attr_reader :metric_snapshots, :metric_store
+
+          def initialize
+            @metric_snapshots = []
+            @mutex = Mutex.new
+            @metric_store = OpenTelemetry::SDK::Metrics::State::MetricStore.new
+          end
+
+          def collect
+            @metric_store.collect
+          end
```

Comment:
I think it should be something like:
```suggestion
        class InMemoryMetricPullExporter < MetricReader
          attr_reader :metric_snapshots

          def initialize
            super
            @metric_snapshots = []
            @mutex = Mutex.new
          end
```